functionary/ui/forms/tasks.py

```diff
@@ -160,3 +165,76 @@ def _get_field_info(
         the field class and widget based on the input data.
         """
         return _field_mapping.get(parameter_type, (None, None))
+
+
+class TaskParameterTemplateForm(TaskParameterForm):
+    """TaskParameterForm variant with template variable support
+
+    This is a version of TaskParameterForm that allows template variables to be entered
+    as values regardless of parameter type. That is, a field requiring an integer will
+    allow either and integer value or a template variable such as {{step.result}}.
+    Other than the exception for template variables, other validation rules still apply.
+    """
+
+    def _stringify_template_variables(self, parameter_template: str) -> str:
+        """Converts the template variables in the parameter template to strings so
+        that the template can be safely jsonified.
+        """
+        return re.sub(r"{{([\w\.]+)}}", r'"{{\1}}"', parameter_template)
+
+    def _build_parameter_template(self, parameters: dict) -> str:
+        """Undo the template variable stringification that was required to jsonify
+        the template. The resulting string is one that can be rendered as a django
+        template"""
+        json_data = json.dumps(parameters)
+
+        return re.sub(r'"{{([\w\.]*)}}"', r"{{\1}}", json_data)
+
+    def __init__(
+        self,
+        function: "Function",
+        data: Optional["QueryDict"] = None,
+        initial: str | None = None,
```

Comment:
I only recently learned that the `|` form was a thing and prefer it since it doesn't require importing stuff from `typing`.  Unfortunately, you can't use that form with types that are only import for typing purposes (the stuff under `if TYPE_CHECKING`).  Thus the use of `Optional["QueryDict"]`.  Since I'm importing `Optional` anyway in this case, I switched to using that in both cases to keep the definitions consistent here.